dataactvalidator/config/sqlrules/b12_object_class_program_activity_1.sql

```diff
@@ -2,21 +2,21 @@ SELECT
     op.row_number,
     op.by_direct_reimbursable_fun
 FROM object_class_program_activity as op
-WHERE submission_id = {} AND (
-    ussgl480100_undelivered_or_fyb IS NOT NULL
-    OR ussgl480100_undelivered_or_cpe IS NOT NULL
-    OR ussgl483100_undelivered_or_cpe IS NOT NULL
-    OR ussgl488100_upward_adjustm_cpe IS NOT NULL
-    OR ussgl490100_delivered_orde_fyb IS NOT NULL
-    OR ussgl490100_delivered_orde_cpe IS NOT NULL
-    OR ussgl493100_delivered_orde_cpe IS NOT NULL
-    OR ussgl498100_upward_adjustm_cpe IS NOT NULL
-    OR ussgl480200_undelivered_or_cpe IS NOT NULL
-    OR ussgl480200_undelivered_or_fyb IS NOT NULL
-    OR ussgl483200_undelivered_or_cpe IS NOT NULL
-    OR ussgl488200_upward_adjustm_cpe IS NOT NULL
-    OR ussgl490200_delivered_orde_cpe IS NOT NULL
-    OR ussgl490800_authority_outl_fyb IS NOT NULL
-    OR ussgl490800_authority_outl_cpe IS NOT NULL
-    OR ussgl498200_upward_adjustm_cpe IS NOT NULL )
-    AND COALESCE(by_direct_reimbursable_fun, '') = ''
+WHERE op.submission_id = {} AND (
+    COALESCE(op.ussgl480100_undelivered_or_fyb, 0) <> 0
+    OR COALESCE(op.ussgl480100_undelivered_or_cpe, 0) <> 0
+    OR COALESCE(op.ussgl483100_undelivered_or_cpe, 0) <> 0
+    OR COALESCE(op.ussgl488100_upward_adjustm_cpe, 0) <> 0
+    OR COALESCE(op.ussgl490100_delivered_orde_fyb, 0) <> 0
+    OR COALESCE(op.ussgl490100_delivered_orde_cpe, 0) <> 0
+    OR COALESCE(op.ussgl493100_delivered_orde_cpe, 0) <> 0
+    OR COALESCE(op.ussgl498100_upward_adjustm_cpe, 0) <> 0
+    OR COALESCE(op.ussgl480200_undelivered_or_cpe, 0) <> 0
+    OR COALESCE(op.ussgl480200_undelivered_or_fyb, 0) <> 0
+    OR COALESCE(op.ussgl483200_undelivered_or_cpe, 0) <> 0
+    OR COALESCE(op.ussgl488200_upward_adjustm_cpe, 0) <> 0
+    OR COALESCE(op.ussgl490200_delivered_orde_cpe, 0) <> 0
+    OR COALESCE(op.ussgl490800_authority_outl_fyb, 0) <> 0
+    OR COALESCE(op.ussgl490800_authority_outl_cpe, 0) <> 0
+    OR COALESCE(op.ussgl498200_upward_adjustm_cpe, 0) <> 0)
+    AND COALESCE(op.by_direct_reimbursable_fun, '') = ''
```

Comment:
No doubt you've thought of this already, @nmonga91 but would also add that we might also want to add the objectclass to the `SELECT` statement so it shows up on the error report.